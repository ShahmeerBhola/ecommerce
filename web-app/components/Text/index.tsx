import React from 'react';

export type Props = {
  className?: string;
  size?: Styles.FontSizes;
  color?: Styles.FontColors;
  weight?: Styles.FontWeights;
  h1?: boolean;
  h2?: boolean;
};

type CalculateClassNameInput = Omit<Props, 'h1' | 'h2'>;

const Text: React.FunctionComponent<Props> = ({ children, className, color, size, weight, h1 = false, h2 = false }) => {
  const calculateClassNames = ({
    color: defaultColor,
    size: defaultSize,
    weight: defaultWeight,
  }: CalculateClassNameInput): string => {
    const col = color || defaultColor;
    const siz = size || defaultSize;
    const weig = weight || defaultWeight;

    return `font-primary ${col} ${siz} ${weig} ${className}`;
  };

  if (h1) {
    return (
      <h1
        className={`${calculateClassNames({ color: 'text-black', size: 'text-5xl', weight: 'font-bold' })} uppercase`}
      >
        {children}
      </h1>
    );
  } else if (h2) {
    return (
      <h2
        className={`${calculateClassNames({
          color: 'text-gray-400',
          size: 'text-4xl',
          weight: 'font-bold',
        })} uppercase`}
      >
        {children}
      </h2>
    );
  }
  return (
    <p className={calculateClassNames({ color: 'text-gray-100', size: 'text-base', weight: 'font-normal' })}>
      {children}
    </p>
  );
};

export default Text;
