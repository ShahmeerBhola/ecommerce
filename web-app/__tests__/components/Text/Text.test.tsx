import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../../../components/Text/Text';

describe('Text component', () => {
  test('it renders properly with default props', () => {
    const tree = renderer.create(<Text>Text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with non-default color', () => {
    const tree = renderer.create(<Text color="red-200">Text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with size prop', () => {
    const tree = renderer.create(<Text size="3xl">Text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with weight prop', () => {
    const tree = renderer.create(<Text weight="extrabold">Text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with className prop', () => {
    const tree = renderer.create(<Text className="my-custom-class">Text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
