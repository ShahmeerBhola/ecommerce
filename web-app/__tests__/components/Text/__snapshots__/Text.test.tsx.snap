// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component it renders properly with className prop 1`] = `
<p
  className="font-secondary font-normal text-gray-100 text-base my-custom-class"
>
  Text
</p>
`;

exports[`Text component it renders properly with default props 1`] = `
<p
  className="font-secondary font-normal text-gray-100 text-base"
>
  Text
</p>
`;

exports[`Text component it renders properly with non-default color 1`] = `
<p
  className="font-secondary font-normal text-red-200 text-base"
>
  Text
</p>
`;

exports[`Text component it renders properly with size prop 1`] = `
<p
  className="font-secondary font-normal text-gray-100 text-3xl"
>
  Text
</p>
`;

exports[`Text component it renders properly with weight prop 1`] = `
<p
  className="font-secondary font-extrabold text-gray-100 text-base"
>
  Text
</p>
`;
